ops/jujucontext.py

```diff
@@ -12,36 +12,84 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""A helper to work with the Juju context and version."""
+"""A helper to work with the Juju context."""
 
 from __future__ import annotations
 
 import dataclasses
+import os
 from pathlib import Path
 from typing import Any, Mapping
 
 from .jujuversion import JujuVersion
 
 
-@dataclasses.dataclass(frozen=True)
-class _JujuContext:
-    """_JujuContext collects information from environment variables named 'JUJU_*'.
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class JujuContext:
+    """JujuContext provides the Juju hook context.
```

Comment:
I think we can drop the class name, as for other classes. The last part of the docstring feels important to reiterate here. How about this:

```suggestion
    """Provides the Juju hook context, primarily for charming approaches outside of Ops.
```